master/internal/db/postgres_trial.go

```diff
@@ -261,6 +261,16 @@ VALUES
 func (db *PgDB) ensureStep(
 	ctx context.Context, tx *sqlx.Tx, trialID, trialRunID, stepsCompleted int,
 ) error {
+	var exists bool
+	if err := tx.QueryRowxContext(ctx, `
+SELECT EXISTS(SELECT 1 FROM steps WHERE trial_id = $1 AND total_batches = $2);`,
+		trialID, stepsCompleted).Scan(&exists); err != nil {
+		return err
+	}
+	if exists {
+		return nil
+	}
+
```

Comment:
style, up for debate, use a switch?

Reply:
Changed to a switch. 

I notice we use switches a lot where I would use if elses and I don't think I really understand the reason? Here I guess it keeps exists coupled with the query statement?

Reply:
yeah, that's basically the reason for me. idk, "go style" (whatever) usually prefers switches to chained if else (if this was an else if instead, the linter would've said the same). part of it for me is just ease of reading code written in the same style.